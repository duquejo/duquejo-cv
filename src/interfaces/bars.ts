export interface Bars {
  color: string;
  tag:   string;
  value: number;
}