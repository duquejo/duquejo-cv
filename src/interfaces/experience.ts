
export interface Experience {
  start_date:      string;
  end_date:        string;
  enterprise:      string;
  isRecent:        boolean;
  project:         string;
  role:            string;
  type:            Type;
  resume:          string;
  additional_info: string;
}

export type Sizes = 'small' | 'large' | 'medium';

export enum Type {
  List = 'list',
  Paragraph = 'paragraph',
}