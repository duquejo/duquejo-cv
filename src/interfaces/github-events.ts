
// Generated by https://quicktype.io

export interface Event {
  id:         string;
  type:       EventType;
  actor:      Actor;
  repo:       EventRepo;
  payload:    Payload;
  public:     boolean;
  created_at: string;
}

export interface Actor {
  id:            number;
  login:         string;
  display_login: string;
  gravatar_id:   string;
  url:           string;
  avatar_url:    string;
}

export interface Payload {
  repository_id?: number;
  push_id?:       number;
  size?:          number;
  distinct_size?: number;
  ref?:           null | string;
  head?:          string;
  before?:        string;
  commits?:       Commit[];
  ref_type?:      string;
  master_branch?: string;
  description?:   string;
  pusher_type?:   string;
  action?:        string;
  number?:        number;
  pull_request?:  PullRequest;
}

export interface Commit {
  sha:      string;
  author:   Author;
  message:  string;
  distinct: boolean;
  url:      string;
}

export interface Author {
  email: string;
  name:  string;
}

export interface PullRequest {
  url:                   string;
  id:                    number;
  node_id:               string;
  html_url:              string;
  diff_url:              string;
  patch_url:             string;
  issue_url:             string;
  number:                number;
  state:                 string;
  locked:                boolean;
  title:                 string;
  user:                  User;
  body:                  null;
  created_at:            string;
  updated_at:            string;
  closed_at:             null | string;
  merged_at:             null | string;
  merge_commit_sha:      null | string;
  assignee:              null;
  assignees:             any[];
  requested_reviewers:   any[];
  requested_teams:       any[];
  labels:                any[];
  milestone:             null;
  draft:                 boolean;
  commits_url:           string;
  review_comments_url:   string;
  review_comment_url:    string;
  comments_url:          string;
  statuses_url:          string;
  head:                  Base;
  base:                  Base;
  _links:                Links;
  author_association:    string;
  auto_merge:            null;
  active_lock_reason:    null;
  merged:                boolean;
  mergeable:             null;
  rebaseable:            null;
  mergeable_state:       string;
  merged_by:             User | null;
  comments:              number;
  review_comments:       number;
  maintainer_can_modify: boolean;
  commits:               number;
  additions:             number;
  deletions:             number;
  changed_files:         number;
}

export interface Links {
  self:            Comments;
  html:            Comments;
  issue:           Comments;
  comments:        Comments;
  review_comments: Comments;
  review_comment:  Comments;
  commits:         Comments;
  statuses:        Comments;
}

export interface Comments {
  href: string;
}

export interface Base {
  label: string;
  ref:   string;
  sha:   string;
  user:  User;
  repo:  BaseRepo;
}

export interface BaseRepo {
  id:                          number;
  node_id:                     string;
  name:                        string;
  full_name:                   string;
  private:                     boolean;
  owner:                       User;
  html_url:                    string;
  description:                 string;
  fork:                        boolean;
  url:                         string;
  forks_url:                   string;
  keys_url:                    string;
  collaborators_url:           string;
  teams_url:                   string;
  hooks_url:                   string;
  issue_events_url:            string;
  events_url:                  string;
  assignees_url:               string;
  branches_url:                string;
  tags_url:                    string;
  blobs_url:                   string;
  git_tags_url:                string;
  git_refs_url:                string;
  trees_url:                   string;
  statuses_url:                string;
  languages_url:               string;
  stargazers_url:              string;
  contributors_url:            string;
  subscribers_url:             string;
  subscription_url:            string;
  commits_url:                 string;
  git_commits_url:             string;
  comments_url:                string;
  issue_comment_url:           string;
  contents_url:                string;
  compare_url:                 string;
  merges_url:                  string;
  archive_url:                 string;
  downloads_url:               string;
  issues_url:                  string;
  pulls_url:                   string;
  milestones_url:              string;
  notifications_url:           string;
  labels_url:                  string;
  releases_url:                string;
  deployments_url:             string;
  created_at:                  string;
  updated_at:                  string;
  pushed_at:                   string;
  git_url:                     string;
  ssh_url:                     string;
  clone_url:                   string;
  svn_url:                     string;
  homepage:                    null;
  size:                        number;
  stargazers_count:            number;
  watchers_count:              number;
  language:                    string;
  has_issues:                  boolean;
  has_projects:                boolean;
  has_downloads:               boolean;
  has_wiki:                    boolean;
  has_pages:                   boolean;
  has_discussions:             boolean;
  forks_count:                 number;
  mirror_url:                  null;
  archived:                    boolean;
  disabled:                    boolean;
  open_issues_count:           number;
  license:                     null;
  allow_forking:               boolean;
  is_template:                 boolean;
  web_commit_signoff_required: boolean;
  topics:                      any[];
  visibility:                  string;
  forks:                       number;
  open_issues:                 number;
  watchers:                    number;
  default_branch:              string;
}

export interface User {
  login:               string;
  id:                  number;
  node_id:             string;
  avatar_url:          string;
  gravatar_id:         string;
  url:                 string;
  html_url:            string;
  followers_url:       string;
  following_url:       string;
  gists_url:           string;
  starred_url:         string;
  subscriptions_url:   string;
  organizations_url:   string;
  repos_url:           string;
  events_url:          string;
  received_events_url: string;
  type:                UserType;
  site_admin:          boolean;
}

export enum UserType {
  User = 'User',
}

export interface EventRepo {
  id:   number;
  name: string;
  url:  string;
}

export enum EventType {
  CreateEvent = 'CreateEvent',
  PullRequestEvent = 'PullRequestEvent',
  PushEvent = 'PushEvent',
}
