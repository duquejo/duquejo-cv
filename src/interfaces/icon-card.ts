
export interface IconCard {
  title: string;
  content: string;
  icon: string;
}

export enum Order {
  Col = 'col',
  Row = 'row'
};