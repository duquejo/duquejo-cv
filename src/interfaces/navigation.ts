export type NavigationLayout = 'menu' | 'fixed';