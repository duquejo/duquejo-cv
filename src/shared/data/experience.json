[
  {
    "date_range": "2023 - Actualidad",
    "enterprise": "SURA Seguros",
    "isRecent": true,
    "project": "SURA - Bus de integraciones",
    "role": "Desarrollador Backend",
    "type": "paragraph",
    "resume": "Diseño y creación de servicios serverless para la nube utilizando Azure DevOps, implementando arquitecturas orientadas a eventos (RabbitMQ / ServiceBus) y servicios de característica HTTP/REST orientados a solucionar todas las integraciones requeridas entre Salesforce Industries y SURA Colombia en el proceso de venta y expedición de seguros para vehículos.",
    "additional_info": "Aptitudes: Azure Functions, Azure Service Bus, CosmoDB, Terraform, Microservicios, Node.js, JMeter, Unit Tests, Integration Tests, SonarQube, Git Flow, JWT, Scrum."
  },
  {
    "date_range": "11/07/2022 - 28/04/2023",
    "enterprise": "Aval Digital Labs - Grupo Aval AV Villas",
    "isRecent": false,
    "project": "Aval Digital Labs - Migración infraestructura Banco AV Villas Tarjeta de crédito",
    "role": "Desarrollador Backend Senior",
    "type": "list",
    "resume": "<li>Migración de infraestructura legada creada manualmente desde Amazon Web Services a recursos de Infraestructura como código usando Serverless Framework y Terraform garantizando estándares de empresa y total supervisión de la configuración de los componentes de la arquitectura de la aplicación.</li><li>Refactorización y optimización de código legado otorgado a lambdas (Node JS/Java) y monolitos (Java Springboot) intervinientes en la aplicación, mejorando hasta un porcentaje mayor o igual al 85% de los estándares Quality Gates frente a buenas prácticas requeridas por la compañía, mejora de rendimiento de ejecución de lambdas y código asociado a Step functions utilizados en el aplicativo.</li><li>Generación de nuevos componentes con lógica de negocio agregada en calidad de apoyo a la célula de trabajo en pararelo a actividades de migración, permitiendo flexibilidad entre las cargas de la migración.</li><li>Generación de políticas de seguridad y correcciones de vulnerabilidades conforme al estándar OWASP, cuyos resultados respondieron adecuadamente a las altas exigencias internas de la organización <i>Aval Digital Labs</i> tanto como al Banco AV Villas.</li>",
    "additional_info": "Aptitudes: DynamoDB, Node.js, Spring Framework, Express, Angular, HTML, AWS, Gradle, Jira, Maven, JMeter, Jenkins, Docker, Unit Test, Functional test, AWS Cognito, Microservicios, Git Flow, JWT."
  },
  {
    "date_range": "07/02/2022 - 06/07/2022",
    "enterprise": "Aval Digital Labs & Colegio los Nogales - MentuApp",
    "isRecent": false,
    "project": "Mentu App",
    "role": "Desarrollador Full-Stack",
    "type": "list",
    "resume": "<li>Desarrollo de una plataforma de aprendizaje para primaria y secundaria.</li><li>Partiendo de una arquitectura hexagonal, orientada a microservicios y usando Serverless Framework con AWS Node JS (Typescript) AWS Lambdas, se favorecieron los principios SOLID, de mantenibilidad y optimización de recursos garantizando una alta disponibilidad y alta persistiencia de la información a través de AWS DocumentDB utilizando Mongoose como ODM para el aplicativo final y TypeORM para bases de datos relacionales AWS RDS de servicios externos Postgres y MySQL.</li><li>Con Serverless framework se disponibilizaron estrategias con AWS Cloudformation para descentralizar cada Lambda para corresponder al Datalake deseado, conectando servicios de Amazon Web Services, tales como API Gateway, S3 (hooks), SNS, SQS, entre otros.</li><li>Se disponibilizó una ETL que entrega información estandarizada y optimizada para la plataforma Metabase, generando a través de AWS Eventbridge información actualizada e íntegra.</li><li>El frontend se construyó con React JS, Webpack y Typescript, dividiendo el core del negocio en diferentes microservicios a través de Module Federation y generalizando componentes gráficos del frontend a través de una librería común de NPM basados en buenas prácticas de la librería.</li><li>Se conectaron aplicaciones externas a través de microservicios como Mixpanel, Concerto, Hotjar, Analytics para garantizar estrategias UX para el público objetivo (Profesores y estudiantes).</li>",
    "additional_info": "Aptitudes: MySQL, MongoDB, PostgreSQL, Nest JS, AWS SQS, Node.js, HTML, CSS, React.js, Tailwind, AWS Cognito, Typescript, Serverless Framework."
  },
  {
    "date_range": "12/12/2021 - 27/12/2021",
    "enterprise": "Ceiba Software S.A.S",
    "isRecent": false,
    "project": "ADN Ceiba",
    "role": "Arquitecto Desarrollador",
    "type": "list",
    "resume": "<li>Se definió y elaboró una arquitectura hexagonal en donde se favorece la orientación al dominio, principios solid y mantenibilidad de los proyectos.</li><li>Todo esto bajo un proceso de integración continua con análisis estático de código y ejecución de pruebas que garantizan la calidad de el proyecto desarrollado.</li><li>Se logro construir front implementado bajo una arquitectura core, share y feature,con manejo de esta global usando redux.</li>",
    "additional_info": "Aptitudes: PostgreSQL, MySQL, SASS, HTML, CSS3, React.js, TypeScript, Redux, Nest JS, JMeter, NPM, Jenkins, Sonar, Git, Unit test, Integration test, E2E Test."
  },
  {
    "date_range": "05/05/2021 - 10/12/2021",
    "enterprise": "SM Digital",
    "isRecent": false,
    "project": "SM Digital - Creación de sitios web e integraciones entre terceros",
    "role": "Programador web líder",
    "type": "list",
    "resume": "<li>Creación de diversos sitios web empleando WordPress como CMS, modularización de funcionalidades de la lógica del negocio a través de buenas prácticas vía Plugins, creación de temas a la medida, totalmente responsivos y optimizados para la web, ideados con el ánimo de optimizar ganancias y generar visibilidad de los clientes a través de estrategias de Marketing.</li><li>Implementación de tiendas virtuales e integraciones con terceros, como pasarelas de pago, ERP, CRM, especializados para la recolección de leads usados en Marketing, además implementación de servicios SSO (Single Sign-On) con redes sociales como Facebook/Twitter/Instagram, entre otros.</li><li>Implementación de estrategias de mensajería automatizada a la medida para plataformas vía WhatsApp, usando APIs autorizadas para el negocio, el cual permitieron comunicar informes a listados de clientes autorizados de manera sencilla.</li><li>Se utilizaron diversas estrategias para el frontend y backend, resaltando HTML, CSS, JS (Frameworks jQuery, Angular JS y React JS), PHP y Node JS respectivamente para la creación de sitios web y generación de aplicaciones para terceros, así como integradores entre plataformas no compatibles.</li>",
    "additional_info": "Aptitudes: MySQL, MongoDB, jQuery, Node.js, Express, React.js, Bootstrap, SASS, HTML5, CSS3, Jira, Trello, WordPress CMS, WooCommerce, JavaScript, TypeScript, PHP, Auth0, JWT, Laravel."
  },
  {
    "date_range": "05/04/2020 - 29/11/2020",
    "enterprise": "Grupo Nutresa - SM Digital",
    "isRecent": false,
    "project": "E-learning Escuela de clientes",
    "role": "Desarrollador Full-Stack",
    "type": "list",
    "resume": "<li>Proyecto Full-Stack desarrollado desde cero, que debía ser construído bajo el framework del CMS WordPress el cual exponía servicios REST para ser consumidos desde el cliente del aplicativo.</li><li>El frontend debía ser construído con Angular JS y se permitía el uso de librerías para la manipulación de contenidos multimedia.</li><li>Implementar estrategias UX debido a la población objetivo (50 años o más)</li><li>Debía optimizar la entrega de los recursos multimedia y responder a un flujo elevado de usuarios, empleando AWS S3, AWS CloudFront, entre otros.</li>",
    "additional_info": "Aptitudes: MySQL, Angular JS, Bootstrap, SASS, HTML, CSS3, AWS, Jira, Trello, Git, Javascript, PHP, MVC."
  },
  {
    "date_range": "14/12/2019 - 04/05/2021",
    "enterprise": "SURA Seguros",
    "isRecent": false,
    "project": "SURA Asesores",
    "role": "Programador Full-Stack",
    "type": "list",
    "resume": "<li>Proyecto Full-Stack donde se requería la integración de un directorio activo SEUS 4 (Asesores de Seguros SURA - SAML) integrándolo con un E-Commerce usando WordPress con WooCommerce como CMS.</li><li>Conforme a las buenas prácticas para WordPress, se requirió montar un plugin adaptable, basado en una arquitectura MVC, el cual favoreció la división de la lógica y componentes del mismo.</li><li>El aplicativo debía contar con reportes a la medida, generados vía CSV el cual provienen de diferentes fuentes tales como Google Docs y REST API externos.</li><li>Se logró culminar el proyecto acatando los requerimientos funcionales y la calidad del software que abogaba el cliente.</li>",
    "additional_info": "Aptitudes: SASS, CSS3, jQuery, Bootstrap, Wordpress, WooCommerce, NPM, Jira, Trello, SAML, PHP, JWT, Azure Active Directory MySQL, Git, JavaScript."
  },
  {
    "date_range": "16/05/2019 - 23/06/2019",
    "enterprise": "Alimentos cárnicos Zenú",
    "isRecent": false,
    "project": "Zenú corporativo - Juego ¡Avanza y aprende!",
    "role": "Programador Frontend",
    "type": "list",
    "resume": "<li>Proyecto Frontend interactivo desarrollado desde cero, con la tecnología Angular 7 bajo la arquitectura recomendada por la guía de estilos de Angular.</li><li>Buscaba satisfacer los principios de mantenibilidad, respetando conceptos de código limpio y patrones de diseño sugeridos por el framework.</li>",
    "additional_info": "Aptitudes: Angular JS, AWS, Jira, Trello, PHP, JavaScript, TypeScript, Functional Test."
  },
  {
    "date_range": "16/05/2019 - 23/06/2019",
    "enterprise": "Grupo Familia",
    "isRecent": false,
    "project": ">Grupo familia - Simulador de servilletas",
    "role": "Programador Full-Stack",
    "type": "list",
    "resume": "<li>Proyecto Full-Stack de carácter interactivo desarrollado desde cero, bajo el Framework del CMS WordPress</li><li>Debido a la infraestructura sugerida por el cliente, era necesario utilizar jQuery y librerías enfocadas para la manipulación del DOM.</li><li>El proyecto exigía respetar directivas muy específicas de la marca directamente relacionadas con los requerimientos funcionales.</li>",
    "additional_info": "Aptitudes: MySQL, Bootstrap, SASS, HTML, Jira, Trello, WordPress, PHP, MVC, jQuery."
  },
  {
    "date_range": "14/12/2019 - 04/05/2021",
    "enterprise": "COLGAS",
    "isRecent": false,
    "project": "Insa (NORGAS) - Cotizador energético",
    "role": "Programador Frontend",
    "type": "list",
    "resume": "<li>Proyecto frontend desarrollado desde cero, que exigía usar enteramente la tecnología JavaScript (Vanilla).</li><li>Dicho cotizador empleaba unos cálculos muy precisos que son cumplidos en detalle debido a políticas de calidad de la compañía, por lo que la interacción con el cliente debía ser constante.</li><li>Se configuró debidamente el proceso de despliegue contínuo mediante tareas programadas un servidor remoto en la nube.</li>",
    "additional_info": "Aptitudes: MySQL, Bootstrap, SASS, HTML, Jira, JavaScript."
  },
  {
    "date_range": "03/02/2019 - 12/06/2021",
    "enterprise": "Grupo Presente Éxito",
    "isRecent": false,
    "project": "Cotizador de centros vacacionales",
    "role": "Programador Full-Stack",
    "type": "list",
    "resume": "<li>Proyecto Full-Stack desarrollado desde cero, con las tecnologías PHP y usando el Framework para páginas web WordPress.</li><li>Conforme a las buenas prácticas del CMS Wordpress, se requería construir enteramente mediante un plugin, implementando una arquitectura MVC.</li><li>Era importante para el cliente responder claramente a los requerimientos funcionales, cuya importancia radicaba en el cálculo, la evaluación de resultados y despliegue de los mismos (PDF - Email), el cual eran utilizadas para estrategias de Marketing y Pauta digital. Era imprescindible que fuera totalmente administrable y configurable.</li>",
    "additional_info": "Aptitudes: MySQL, Bootstrap, SASS, HTML, CSS, WordPress, Jira, JavaScript, MVC, Trello, PHP, jQuery."
  },
  {
    "date_range": "07/11/2017 - 10/10/2018",
    "enterprise": "Universidad EAFIT",
    "isRecent": false,
    "project": "Universidad EAFIT - Plataforma Línea en Informática Educativa (Conexa EAFIT)",
    "role": "Programador Full-Stack",
    "type": "list",
    "resume": "<li>Proyecto Full-Stack desarrollado desde cero, bajo el Framework del CMS WordPress (PHP).</li><li>Requería emplear buenas prácticas del CMS para su construcción, por lo que sus módulos se desarrollaron bajo la arquitectura Kernel/Plugin, común en CMS como WordPress.</li><li>Requería manejar información dinámicamente desde el frontend usando jQuery y Ajax.</li><li>Debían mantenerse diferentes estructuras de servicios en la nube para sus usuarios y colaboradores, empleando la suite de AWS para ello. (S3, Glacier, Cloudfront, Route 53, etc) y administración de servicios Cloud de Azure y Office 365.</li>",
    "additional_info": "Aptitudes: MySQL, Bootstrap, SASS, HTML, CSS, WordPress, Jira, JavaScript, AWS, MVC, Trello, PHP, jQuery, Office 365."
  }
]