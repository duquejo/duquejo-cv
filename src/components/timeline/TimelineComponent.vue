<template>
  <ol class="pt-1 relative border-s border-gray-200">
    <li class="cursor-auto mb-8 ms-4">
      <div data-test="detail" class="absolute w-2 h-2 rounded-full mt-1.5 -start-1" :class="currentDetail" />

      <!-- Aligned content -->
      <div class="flex items-center justify-between lg:justify-normal">
        <span data-test="time" class="lg:inline-flex border-2 rounded px-2 py-1 font-semibold text-xs text-center" :class="dateClass">
          <time v-if="start_date" class="block" v-text="start_date" />
          <time v-if="end_date" class="block mt-1 lg:mt-0 lg:before:content-['-'] lg:before:mx-1" v-text="end_date" />
        </span>
        <span class="text-black text-xs font-normal ml-2 text-right lg:text-left" v-text="enterprise" />
      </div>

      <!-- Meta -->
      <span class="block text-xs mt-3 font-semibold mb-3 lg:mb-0">{{ project }}</span>
      <span class="block text-base font-semibold">{{ role }}</span>

      <!-- Content -->
      <p v-if="type === Type.Paragraph " class="my-1 text-justify" v-html="resume"/>
      <ul v-else-if="type === Type.List" class="my-1 list-disc ml-4 space-y-2 text-justify" v-html="resume"/>

      <!-- Tags -->
      <small class="block font-semibold mt-2 text-black text-justify" v-text="additional_info" />
    </li>
  </ol>
</template>

<script setup lang="ts">
import { toRefs, computed } from 'vue';
import { Experience, Type } from '../../interfaces/experience';

const props = defineProps<Experience>();
const { additional_info, start_date, end_date, enterprise, isRecent, project, role, type, resume } = toRefs(props);

const dateClass = computed(() => ({
  'border-gray-400': ! isRecent.value,
  'border-yellow-400': isRecent.value,
}));

const currentDetail = computed(() => ({
  'bg-gray-400': ! isRecent.value,
  'bg-yellow-400': isRecent.value,
}));

</script>