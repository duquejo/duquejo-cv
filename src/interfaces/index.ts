import type { Bars } from './bars';
import type { Experience } from './experience';
import type { IconCard } from './icon-card';
import { Type } from './experience';
import { Order } from './icon-card';


export { Bars, Experience, IconCard, Order, Type };