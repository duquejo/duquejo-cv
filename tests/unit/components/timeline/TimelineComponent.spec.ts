import { shallowMount } from '@vue/test-utils';
import { describe, expect, it, vi, beforeEach } from 'vitest';
import TimelineComponent from '../../../../src/components/timeline/TimelineComponent.vue';
import { TimelineBuilder } from '../../../utils/TimelineBuilder';
import { Experience } from '../../../../src/interfaces/experience';

describe('+ TimelineComponent.vue tests', () => {

  const generateWrapper = (timeline?: Experience) => 
    shallowMount(TimelineComponent, {
      props: timeline && { ...timeline },
    }
  );

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('Should match the snapshot - Paragraph', () => {
    const timeline = new TimelineBuilder().withParagraph().build();
    const wrapper = generateWrapper(timeline);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should match the snapshot - List', () => {
    const timelineList = new TimelineBuilder().withList().build();
    const wrapper = generateWrapper(timelineList);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should map the properties as expected with the paragraph type', () => {
    const timeline = new TimelineBuilder().withParagraph().build();
    const wrapper = generateWrapper(timeline);

    const time = wrapper.find('time');
    const content = wrapper.find('p');
    const small = wrapper.find('small');
    
    expect(time.text()).toBe(timeline.date_range);
    expect(content.text()).toBe(timeline.resume);
    expect(small.text()).toBe(timeline.additional_info);

    expect(time.classes()).toContain('border-gray-400');
    expect(time.classes()).not.toContain('border-yellow-500');
  });

  it('should map the properties as expected with the list type', () => {
    const timelineList = new TimelineBuilder().withList().build();
    const wrapper = generateWrapper(timelineList);

    const detail = wrapper.find('.timeline--current-detail');
    const time = wrapper.find('time');
    const content = wrapper.findAll('ul li');
    const small = wrapper.find('small');
    
    expect(time.text()).toBe(timelineList.date_range);
    expect(content).toHaveLength(2);
    expect(small.text()).toBe(timelineList.additional_info);

    expect(detail.classes()).toContain('bg-gray-400');
    expect(detail.classes()).not.toContain('bg-yellow-500');
  });

  it('should conditionate date classes - Recent date', () => {
    const timelineList = new TimelineBuilder().withParagraph().setIsRecent(true).build();
    const wrapper = generateWrapper(timelineList);

    const time = wrapper.find('time');

    expect(time.classes()).toContain('border-yellow-500');
    expect(time.classes()).not.toContain('border-gray-400');
  });

  it('should conditionate detail classes - Recent detail', () => {
    const timelineList = new TimelineBuilder().withParagraph().setIsRecent(true).build();
    const wrapper = generateWrapper(timelineList);

    const detail = wrapper.find('.timeline--current-detail');

    expect(detail.classes()).toContain('bg-yellow-500');
    expect(detail.classes()).not.toContain('bg-gray-400');
  });
});