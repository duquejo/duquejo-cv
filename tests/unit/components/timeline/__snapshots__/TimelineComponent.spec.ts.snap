// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ TimelineComponent.vue tests > Should match the snapshot - List 1`] = `
"<ol class="pt-1 relative border-s border-gray-200">
  <li class="cursor-auto mb-8 ms-4">
    <div class="timeline--current-detail bg-gray-400"></div>
    <div class="flex items-center justify-between lg:justify-normal"><time class="timeline--time border-gray-400">11/07/2022 - 28/04/2023</time><span class="text-black text-xs font-normal ml-2 text-right lg:text-left">Foo bar SAS</span></div><span class="block text-xs mt-3 font-semibold mb-3 lg:mb-0">Foo Bar SAS - Project</span><span class="block text-base font-semibold">Senior backend developer</span><!-- Content -->
    <ul class="my-1 list-disc ml-4 space-y-2 text-justify">
      <li>foo bar</li>
      <li>lorem ipsum sit amet.</li>
    </ul><small class="block font-semibold mt-2 text-black text-justify">Skills: Git Flow, JWT.</small>
  </li>
</ol>"
`;

exports[`+ TimelineComponent.vue tests > Should match the snapshot - Paragraph 1`] = `
"<ol class="pt-1 relative border-s border-gray-200">
  <li class="cursor-auto mb-8 ms-4">
    <div class="timeline--current-detail bg-gray-400"></div>
    <div class="flex items-center justify-between lg:justify-normal"><time class="timeline--time border-gray-400">2023 - Present</time><span class="text-black text-xs font-normal ml-2 text-right lg:text-left">John Doe Co</span></div><span class="block text-xs mt-3 font-semibold mb-3 lg:mb-0">John Doe Company Project</span><span class="block text-base font-semibold">Backend developer</span><!-- Content -->
    <p class="my-1 text-justify">foo bar, lorem ipsum sit amet-</p><small class="block font-semibold mt-2 text-black text-justify">Skills: Azure Functions, Azure Service Bus.</small>
  </li>
</ol>"
`;
